okapi-core/src/test/java/org/folio/okapi/InstallTest.java

```diff
@@ -1514,4 +1519,353 @@ public void installTenantInitVersion2JobError(TestContext context) {
+        .add(new JsonObject()
+            .put("id", "a")
+            .put("version", "1.0")
+        )
+        .add(new JsonObject()
+            .put("id", "b")
+            .put("version", "1.0")
+        );
+    given()
+        .header("Content-Type", "application/json")
+        .body(mid.encode()).post("/_/proxy/modules").then().statusCode(201);
+
+    final String tenant = "roskilde";
+    createTenant(context, tenant);
+    deployAsyncInitModule(context, moda.getString("id"), portModule);
+    deployAsyncInitModule(context, modb.getString("id"), portModule);
+    deployAsyncInitModule(context, mid.getString("id"), portModule);
+    ModuleTenantInitAsync tModule = new ModuleTenantInitAsync(vertx, portModule);
+    tModule.start().onComplete(context.asyncAssertSuccess());
+
+    JsonArray installOp = new JsonArray()
+        .add(new JsonObject()
+            .put("id", moda.getString("id"))
+            .put("action", "enable")
+        )
+        .add(new JsonObject()
+            .put("id", modb.getString("id"))
+            .put("action", "enable")
+        )
+        .add(new JsonObject()
+            .put("id", mid.getString("id"))
+            .put("action", "enable")
+        );
+
+    given()
+        .header("Content-Type", "application/json")
+        .body(installOp.encode())
+        .post("/_/proxy/tenants/" + tenant + "/install?async=true&parallel=0")
+        .then().statusCode(400)
+        .body(containsString("parallel must be 1 or higher"));
+
+    RestAssuredClient c = api.createRestAssured3();
+    String location = c.given()
+        .header("Content-Type", "application/json")
+        .body(installOp.encode())
+        .post("/_/proxy/tenants/" + tenant + "/install?async=true&parallel=5")
+        .then().statusCode(201)
+        .extract().header("Location");
+    Assert.assertTrue(
+        "raml: " + c.getLastReport().toString(),
+        c.getLastReport().isEmpty());
+
+    JsonObject job = pollCompleteStrip(context, location);
+    JsonObject jobExpected = new JsonObject()
+        .put("complete", true)
+        .put("modules", new JsonArray()
+            .add(new JsonObject()
+                .put("id", moda.getString("id"))
+                .put("action", "enable")
+                .put("stage", "done")
+            )
+            .add(new JsonObject()
+                .put("id", modb.getString("id"))
+                .put("action", "enable")
+                .put("stage", "done")
+            )
+            .add(new JsonObject()
+                .put("id", mid.getString("id"))
+                .put("action", "enable")
+                .put("stage", "done")
+            )
+        );
+    context.assertEquals(jobExpected, job);
+    // moda , modb running in parallel
+    context.assertTrue(tModule.getStartTime(moda.getString("id")).isBefore(tModule.getEndTime(modb.getString("id"))));
+    context.assertTrue(tModule.getStartTime(modb.getString("id")).isBefore(tModule.getEndTime(moda.getString("id"))));
```

Comment:
`assertParallelRun(ModuleTenantInitAsync a, ModuleTenantInitAsync b)` and
`assertSequentialRun(ModuleTenantInitAsync a, ModuleTenantInitAsync b)`
can avoid duplicate code.

Reply:
ok